// Generated by Haxe 4.0.0-preview.4+1e3e5e016
package utest;

import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public class TestHandler_addEvent_179__Fun<T, EventArg> extends haxe.lang.Function
{
	public TestHandler_addEvent_179__Fun(utest.TestHandler<T> handler, haxe.lang.Function f)
	{
		//line 179 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		super(1, 0);
		//line 179 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		this.handler = handler;
		//line 179 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		this.f = f;
	}
	
	
	@Override public java.lang.Object __hx_invoke1_o(double __fn_float1, java.lang.Object __fn_dyn1)
	{
		//line 179 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		EventArg e = ( (( __fn_dyn1 == haxe.lang.Runtime.undefined )) ? (((EventArg) (((java.lang.Object) (__fn_float1) )) )) : (((EventArg) (__fn_dyn1) )) );
		//line 180 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		if ( ! (this.handler.asyncStack.remove(this.f)) ) 
		{
			//line 181 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			this.handler.results.add(utest.Assertation.AsyncError("event already executed", new haxe.root.Array<haxe.StackItem>(new haxe.StackItem[]{})));
			//line 182 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			return null;
		}
		
		//line 184 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		try 
		{
			//line 185 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			this.handler.bindHandler();
			//line 186 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			this.f.__hx_invoke1_o(0.0, e);
		}
		catch (java.lang.Throwable catchallException)
		{
			//line 184 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			haxe.lang.Exceptions.setException(catchallException);
			//line 188 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			java.lang.Object realException = ( (( catchallException instanceof haxe.lang.HaxeException )) ? (((haxe.lang.HaxeException) (catchallException) ).obj) : (catchallException) );
			//line 188 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
			{
				//line 188 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
				java.lang.Object e1 = realException;
				//line 188 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
				this.handler.results.add(utest.Assertation.AsyncError(e1, utest.TestHandler.exceptionStack(0)));
			}
			
		}
		
		
		//line 179 "C:\\Users\\VEgorov\\haxelib\\utest\\1,8,4\\src\\utest\\TestHandler.hx"
		return null;
	}
	
	
	public utest.TestHandler<T> handler;
	
	public haxe.lang.Function f;
	
}


